curiosrc/alertmanager/task_alert.go

```diff
@@ -0,0 +1,220 @@
+// Nobody associated with this software's development has any business relationship to pagerduty.
+// This is provided as a convenient trampoline to SP's alert system of choice.
+
+package alertmanager
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/lib/harmony/harmonydb"
+	"github.com/filecoin-project/lotus/lib/harmony/harmonytask"
+	"github.com/filecoin-project/lotus/lib/harmony/resources"
+	"github.com/filecoin-project/lotus/node/config"
+	"github.com/filecoin-project/lotus/storage/ctladdr"
+)
+
+const AlertMangerInterval = time.Hour
+
+var log = logging.Logger("curio/alertmanager")
+
+type AlertAPI interface {
+	ctladdr.NodeApi
+	StateMinerInfo(ctx context.Context, actor address.Address, tsk types.TipSetKey) (api.MinerInfo, error)
+}
+
+type AlertTask struct {
+	api AlertAPI
+	cfg config.CurioAlerting
+	db  *harmonydb.DB
+}
+
+type alertOut struct {
+	err         error
+	alertString string
+}
+
+type alerts struct {
+	ctx      context.Context
+	api      AlertAPI
+	db       *harmonydb.DB
+	cfg      config.CurioAlerting
+	alertMap map[string]*alertOut
+}
+
+type pdPayload struct {
+	Summary       string      `json:"summary"`
+	Severity      string      `json:"severity"`
+	Source        string      `json:"source"`
+	Component     string      `json:"component,omitempty"`
+	Group         string      `json:"group,omitempty"`
+	Class         string      `json:"class,omitempty"`
+	CustomDetails interface{} `json:"custom_details,omitempty"`
+}
+
+type pdData struct {
```

Comment:
please just move this inside of sendAlert(). It doesn't go anywhere else so it may-as-well be local.

Reply:
Moved